client/trino-client/src/main/java/io/trino/client/uri/HttpClientFactory.java

```diff
@@ -157,6 +124,41 @@ public static OkHttpClient.Builder unauthenticatedClientBuilder(TrinoUri uri, St
         setupHttpProxy(builder, uri.getHttpProxy());
         setupTimeouts(builder, toIntExact(uri.getTimeout().toMillis()), TimeUnit.MILLISECONDS);
         setupHttpLogging(builder, uri.getHttpLoggingLevel());
+
+        if (!uri.isUseSecureConnection()) {
+            setupInsecureSsl(builder);
+        }
+
+        if (uri.isUseSecureConnection()) {
```

Comment:
nit: Can't we combine the two if conditions to if-else?

Reply:
Yeah. I'll push another commit.